### PR TITLE
fix(ingest/postgres): fix profiling errors, skip json type column

### PR DESCRIPTION
**Pull request by mayurinehate**
## Checklist

- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/ge_data_profiler.py [hunk 1]
```diff
@@ -273,19 +274,32 @@ def _get_columns_to_profile(self) -> List[str]:
...
+
         # Compute ignored columns
-        ignored_columns: List[str] = []
-        for col in self.dataset.get_table_columns():
+        ignored_columns_by_pattern: List[str] = []
+        ignored_columns_by_type: List[str] = []
+
+        types_to_ignore = _get_column_types_to_ignore(self.dataset.engine.dialect.name)
+
+        for col_dict in self.dataset.columns:
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/ge_data_profiler.py:284]: were we previously calling `get_table_columns` twice? and now we're using a cached result instead?

File: metadata-ingestion/src/datahub/ingestion/source/ge_data_profiler.py [hunk 2]
```diff
@@ -1085,3 +1099,13 @@ def _get_ge_dataset(
                 logger.debug(f"Setting table name to be {batch._table}")
 
         return batch
+
+
+# More dialect specific types to ignore can be added here
+# Stringified types are used to avoid dialect specific import errors
+@lru_cache(maxsize=1)
+def _get_column_types_to_ignore(dialect_name: str) -> List[str]:
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/ge_data_profiler.py:1107]: let's change this to _should_ignore_column(dialect_name, type_str) -> bool

in the future we'll run into weird `struct<...>` and things, and so letting this function handle it is easier

**Review by hsheth2 — COMMENTED** [review]
lgtm with a nit

File: metadata-ingestion/src/datahub/ingestion/source/ge_data_profiler.py (hunk 1 above)
Comment (mayurinehate) [reply metadata-ingestion/src/datahub/ingestion/source/ge_data_profiler.py:284]: Earlier, get_table_columns was returning from self.columns (cached result)only,  but it returned only column names (List[str]).

File: metadata-ingestion/src/datahub/ingestion/source/ge_data_profiler.py (hunk 2 above)
Comment (mayurinehate) [reply metadata-ingestion/src/datahub/ingestion/source/ge_data_profiler.py:1107]: sure. One point to note is the type_str here is not native type but sqlalchemy type. This is description of `type` before stringification - 

```
* ``type`` - the type of this column; an instance of
          :class:`~sqlalchemy.types.TypeEngine`
```

**Review by hsheth2 — APPROVED** [review]

**Pull request merged**